File: remove_bg.py
from PIL import Image

def make_white_transparent(image_path):
    img = Image.open(image_path).convert("RGBA")
    datas = img.getdata()
    new_data = []
    for item in datas:
        # If the pixel is close to white, make it transparent
        if item[0] > 240 and item[1] > 240 and item[2] > 240:
            new_data.append((255, 255, 255, 0))
        else:
            new_data.append(item)
    img.putdata(new_data)
    img.save(image_path, "PNG")

make_white_transparent("c:/myportfolio/public/assets/languages/mysql.png")
print("Successfully made mysql.png background transparent.")

File: remove_bg.js
const { Jimp } = require('jimp');

async function removeWhiteBg(imagePath) {
    console.log("Loading image:", imagePath);
    try {
        const image = await Jimp.read(imagePath);
        
        // Loop through each pixel
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function (x, y, idx) {
            const red   = this.bitmap.data[idx + 0];
            const green = this.bitmap.data[idx + 1];
            const blue  = this.bitmap.data[idx + 2];
            
            if (red > 240 && green > 240 && blue > 240) {
                this.bitmap.data[idx + 3] = 0;
            }
        });

        image.write(imagePath);
        console.log("Successfully made background transparent for", imagePath);
    } catch (err) {
        console.error("Error processing image:", err);
    }
}

removeWhiteBg('C:/myportfolio/public/assets/languages/mysql.png');
